redfish/provider/resource_redfish_bios.go

```diff
@@ -1,5 +1,472 @@
+}
+
+// Schema defines the schema for the resource.
+func (*BiosResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This Terraform resource is used to manage user entity of the iDRAC Server." +
+			"We can create, read, modify and delete an existing user using this resource.",
+		Description: "This Terraform resource is used to manage user entity of the iDRAC Server." +
+			"We can create, read, modify and delete an existing user using this resource.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "The ID of the resource.",
+				Description:         "The ID of the resource.",
+				Computed:            true,
+			},
+			"attributes": schema.MapAttribute{
+				MarkdownDescription: "The Bios attribute map.",
+				Description:         "The Bios attribute map.",
+				ElementType:         types.StringType,
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.Map{
+					mapplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"settings_apply_time": schema.StringAttribute{
+				Optional: true,
+				Description: "The time when the BIOS settings can be applied. Applicable value is 'OnReset' only. " +
+					"In upcoming releases other apply time values will be supported. Default is \"OnReset\".",
+				Validators: []validator.String{
+					stringvalidator.OneOf([]string{
+						string(redfishcommon.OnResetApplyTime),
+					}...),
+				},
+				Default:  stringdefault.StaticString(string(redfishcommon.OnResetApplyTime)),
+				Computed: true,
+			},
+			"reset_type": schema.StringAttribute{
+				Optional: true,
+				Description: "Reset type to apply on the computer system after the BIOS settings are applied. " +
+					"Applicable values are 'ForceRestart', " +
+					"'GracefulRestart', and 'PowerCycle'." +
+					"Default = \"GracefulRestart\". ",
+				Validators: []validator.String{
+					stringvalidator.OneOf([]string{
+						string(redfish.ForceRestartResetType),
+						string(redfish.GracefulRestartResetType),
+						string(redfish.PowerCycleResetType),
+					}...),
+				},
+				Computed: true,
+				Default:  stringdefault.StaticString(string(redfish.GracefulRestartResetType)),
+			},
+			"reset_timeout": schema.Int64Attribute{
+				Optional:    true,
+				Description: "reset_timeout is the time in seconds that the provider waits for the server to be reset before timing out.",
+				Default:     int64default.StaticInt64(int64(defaultBiosConfigServerResetTimeout)),
+				Computed:    true,
+			},
+			"bios_job_timeout": schema.Int64Attribute{
+				Optional: true,
+				Description: "bios_job_timeout is the time in seconds that the provider waits for the bios update job to be" +
+					"completed before timing out.",
+				Default:  int64default.StaticInt64(int64(defaultBiosConfigJobTimeout)),
+				Computed: true,
+			},
+		},
+		Blocks: map[string]schema.Block{
```

Comment:
Can we use this function? https://github.com/dell/terraform-provider-redfish/blob/dev/redfish/provider/resource_redfish_power.go#L117